modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/data_set/filters/modals/selection_filter/SelectionFilter.tsx

```diff
@@ -426,6 +438,16 @@ function Body({
 		}
 	}, [preselectedValueInput, source, sourceType]);
 
+	useEffect(() => {
+		if (selectedRESTApplication && selectedRESTEndpoint) {
+			setSource(
+				`/o${selectedRESTApplication.replace('v1.0/', '')}${selectedRESTEndpoint}`
+			);
+
+			setSourceValidationError(false);
+		}
+	}, [selectedRESTApplication, selectedRESTEndpoint]);
```

Comment:
Can we avoid using `useEffect` here? We should be able to update source and validation on events that change selections. If there is a lot of code duplication, we could have a small private util.

Reply:
Done!